src/core/api/plug_scanners_test.go

```diff
@@ -0,0 +1,384 @@
+	m.On("GetRegistrationByProject", 1).Return(r, nil)
+}
+
+// TearDownSuite clears testing env
+func (suite *ScannerAPITestSuite) TearDownSuite() {
+	// Restore
+	api.DefaultController = suite.originC
+}
+
+// TearDownTest clears test case env
+func (suite *ScannerAPITestSuite) TearDownTest() {
+	// List
+	l := make([]*scanner.Registration, 0)
+	err := handleAndParse(&testingRequest{
+		url:        rootRoute,
+		method:     http.MethodGet,
+		credential: sysAdmin,
+	}, &l)
+	require.NoError(suite.T(), err)
+
+	// Clear
+	if len(l) > 0 {
+		for _, r := range l {
+			runCodeCheckingCases(suite.T(), &codeCheckingCase{
+				request: &testingRequest{
+					url:        fmt.Sprintf("%s/%s", rootRoute, r.UUID),
+					method:     http.MethodDelete,
+					credential: sysAdmin,
+				},
+				code: http.StatusOK,
+			})
+		}
+	}
+}
+
+// TestScannerAPICreate tests the post request to create new one
+func (suite *ScannerAPITestSuite) TestScannerAPICreate() {
+	// Including general cases
+	cases := []*codeCheckingCase{
+		// 401
+		{
+			request: &testingRequest{
+				url:    rootRoute,
+				method: http.MethodPost,
+			},
+			code: http.StatusUnauthorized,
+		},
+		// 403
+		{
+			request: &testingRequest{
+				url:        rootRoute,
+				method:     http.MethodPost,
+				credential: nonSysAdmin,
+			},
+			code: http.StatusForbidden,
+		},
+		// 400
+		{
+			request: &testingRequest{
+				url:        rootRoute,
+				method:     http.MethodPost,
+				credential: sysAdmin,
+				bodyJSON: &scanner.Registration{
+					URL: "http://a.b.c",
+				},
+			},
+			code: http.StatusBadRequest,
+		},
+		//201
+		{
+			request: &testingRequest{
+				url:        rootRoute,
+				method:     http.MethodPost,
+				credential: sysAdmin,
+				bodyJSON:   suite.sample,
+			},
+			code: http.StatusCreated,
+		},
+		// 409
+		{
+			request: &testingRequest{
+				url:        rootRoute,
+				method:     http.MethodPost,
+				credential: sysAdmin,
+				bodyJSON:   suite.sample,
+			},
+			code: http.StatusConflict,
+		},
+	}
+
+	runCodeCheckingCases(suite.T(), cases...)
+}
+
+// TestScannerAPIGetAndList tests the get and update API
+func (suite *ScannerAPITestSuite) TestScannerAPIGetAndUpdate() {
+	uid := suite.create()
+
+	// Get
+	r := &scanner.Registration{}
+	err := handleAndParse(&testingRequest{
+		url:        fmt.Sprintf("%s/%s", rootRoute, uid),
+		method:     http.MethodGet,
+		credential: sysAdmin,
+	}, r)
+	require.NoError(suite.T(), err)
+	require.NotNil(suite.T(), r)
+	assert.Equal(suite.T(), "apiTest", r.Name)
+
+	// Update
+	r.Name = "apiTestUpdate"
+	r.Disabled = true
+
+	rr := &scanner.Registration{}
+	err = handleAndParse(&testingRequest{
+		url:        fmt.Sprintf("%s/%s", rootRoute, uid),
+		method:     http.MethodPut,
+		credential: sysAdmin,
+		bodyJSON:   r,
+	}, rr)
+	require.NoError(suite.T(), err)
+	require.NotNil(suite.T(), rr)
+
+	assert.Equal(suite.T(), "apiTestUpdate", rr.Name)
+	assert.Equal(suite.T(), true, rr.Disabled)
```

Comment:
Suite object has assertion methods, `suite.Equal(true, rr.Disabled)`

Reply:
Reply together with the above comment, `require` will exit and assert will not. So, need different methods for the different cases.

Let's keep them as they were.